qiskit_ibm_runtime/runtime_job.py

```diff
@@ -222,29 +218,20 @@ def error_message(self) -> Optional[str]:
         self._set_status_and_error_message()
         return self._error_message
 
-    def wait_for_final_state(
-        self, timeout: Optional[float] = None, wait: float = 5
-    ) -> None:
+    def wait_for_final_state(self, timeout: Optional[float] = None) -> None:
         """Poll the job status until it progresses to a final state such as ``DONE`` or ``ERROR``.
```

Comment:
"Poll the job status" is no longer accurate. Can you shortly document how the code determines when the job reaches a final state?

I don't understand this looking at the code changes alone. Maybe there are some underlying implicit assumptions (e.g. `/stream/jobs/{id}` does not open a stream until a job succeeds) that are not obviously clear.